loadgen/test_settings_internal.cc

```diff
@@ -245,4 +249,122 @@ void TestSettingsInternal::LogSummary(AsyncSummary &summary) const {
 }
 
 }  // namespace loadgen
+
+
+int TestSettings::FromConfig(const std::string &path, const std::string &model,
+                             const std::string &scenario) {
+  std::map<std::string, std::string> kv;
+
+  // lookup key/value pairs from config
+  auto lookupkv = [&](const std::string &model, const std::string &scenario,
+                      const std::string &key, size_t* val_l, double* val_d, double multiplier=1.0) {
+    std::map<std::string, std::string>::iterator it;
+    std::string found;
+    // lookup exact key first
+    it = kv.find(model + scenario + "." + key);
+    if (it != kv.end()) {
+      found = it->second;
+    } else {
+    // lookup key with model wildcard
+    it = kv.find("*." + scenario + "." + key);
+      if (it != kv.end()) {
+        found = it->second;
+      } else {
+        return false;
+      }
+    }
+    // if we get here, found will be set
+    if (val_l) *val_l = strtoul(found.c_str(), 0, 0) * int(multiplier);
+    if (val_d) *val_d = strtod(found.c_str(), 0) * multiplier;
```

Comment:
nit: For strtoul and strtod here and below, please use nullptr, rather than 0.

Reply:
done